src/coreclr/jit/lclmorph.cpp

```diff
@@ -988,7 +988,7 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 {
                     isWide = endOffset.Value() > m_compiler->lvaLclExactSize(lclNum);
 
-                    if (varDsc->TypeGet() == TYP_BLK)
+                    if ((varDsc->TypeGet() == TYP_STRUCT) && varDsc->GetLayout()->IsBlockLayout())
```

Comment:
@SingleAccretion Do you think removing this requires any downstream changes now that these are `TYP_STRUCT`?
I'll probably try that in a follow-up.

Reply:
> Do you think removing this requires any downstream changes now that these are `TYP_STRUCT`?

I wouldn't expect any.